src/screens/EventsScreen/FilterHeader.js

```diff
@@ -13,38 +13,59 @@ import {
 } from "../../constants/colors";
 import text from "../../constants/text";
 import { formatDateRange } from "../../data/formatters";
+import ResetAllFiltersButton from "./ResetAllFiltersButton";
 
 export type Props = {
   onFilterCategoriesPress: Function,
   dateFilter: ?DateRange,
   onFilterButtonPress: () => void,
   onDateFilterButtonPress: () => void,
   selectedCategories: Set<EventCategoryName>,
-  numTagFiltersSelected: number
+  numTagFiltersSelected: number,
+  resetAllFiltersPress: () => void,
+  scrollEventListToTop: () => void
 };
 
 class FilterHeader extends React.PureComponent<Props> {
+  static defaultProps = {
+    resetAllFiltersPress: () => {}
+  };
+
   render() {
     const {
       dateFilter,
       onFilterCategoriesPress,
       selectedCategories,
       onFilterButtonPress,
       onDateFilterButtonPress,
-      numTagFiltersSelected
+      numTagFiltersSelected,
+      resetAllFiltersPress,
+      scrollEventListToTop
     } = this.props;
     const formattedDateFilter = dateFilter
       ? formatDateRange(dateFilter)
       : text.selectDates;
 
+    const anyAppliedFilters: boolean =
+      !!dateFilter || numTagFiltersSelected > 0 || selectedCategories.size > 0;
+
     return (
       <View accessibilityTraits={["header"]} style={styles.container}>
         <ContentPadding>
-          <View testID="event-filter-header" style={styles.content}>
-            <FilterHeaderCategories
-              onFilterPress={onFilterCategoriesPress}
-              selectedCategories={selectedCategories}
+          <View>
+            <ResetAllFiltersButton
+              visible={anyAppliedFilters}
+              onPress={() => {
```

Comment:
If you make `ResetAllFiltersButton` a pure component, this shouldn't be an inline function.

Reply:
Could you elaborate a bit? Not sure I follow this point.

Reply:
As fas as I know, this will generate a new function every time in `render`, which means that inside `ResetAllFiltersButton` this check will always be false: `props.onPress === nextProps.onPress`. So it will render every time.

We could solve this by adding the function to the `FilterHeader` class:

```js
resetAllFilters = () => {
  this.props.resetAllFiltersPress();
  this.props.scrollEventListToTop();
};
```

Then use `onPress={this.resetAllFilters}`. This will always pass in the same function.